### PR TITLE
Migrate helm charts from opensearch-devops repo

### LINKED ISSUE #2: [Migration] Migrate Helm-Charts from opensearch-devops repo to this repo
https://github.com/opensearch-project/opensearch-devops/issues/53

### PR DESCRIPTION
**Pull request by TheAlgo**
### Description
* Migrate the helm charts from [opensearch-devops](https://github.com/opensearch-project/opensearch-devops/tree/main/Helm) to this repo exclusive for Helm charts.
* Change the default config for OpenSearch Dashboards.
 
### Issues Resolved
#2 
[#53](https://github.com/opensearch-project/opensearch-devops/issues/53)
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
@DandyDeveloper @mprimeaux @smlx

**Comment by TheAlgo** [comment]
For #5 I was thinking if we can remove the demo configuration entirely and use the default values of the docker image for OpenSearch Dashboards and comment out the TLS configuration something like this : 

```
config: {}
    ## Default OpenSearch Dashboards configuration from Docker image of Dashboards
    #opensearch_dashboards.yml;
      # server.name: dashboards
      # server.host: "0"
  
  
      ## Dashboards TLS Config (Ensure the cert files are present before enabling SSL
      # server.ssl.enabled: true
      # server.ssl.key: /usr/share/opensearch-dashboards/certs/dashboards.pem
      # server.ssl.certificate: /usr/share/opensearch-dashboards/certs/dashboards-crt.pem
      # elasticsearch.ssl.certificateAuthorities: //usr/share/opensearch-dashboards/certs/dashboards-root-ca.pem
  
      # opensearch_security.cookie.secure: true
      # opensearch_security.cookie.password: ${COOKIE_PASS}

```

**Comment by TheAlgo** [comment]
@peterzhuamazon @DandyDeveloper @mprimeaux @smlx Please review this migration whenever you have time. It is just a copy from the previous repo and changing the demo config for dashboards. Thanks

**Review by DandyDeveloper — COMMENTED** [review]
LGTM. I saw some stuff that needs changing, but nothing critical. 

Good to merge from my perspective.

**Review by DandyDeveloper — APPROVED** [review]

**Comment by mprimeaux** [comment]
Agree with @DandyDeveloper in that LGTM with nothing critical missing. I’ll make it a priority to get a PR for Chart Releaser and Chart Testing once folks are comfortable with this PR.

**Review by mprimeaux — APPROVED** [review]
LGTM

File: charts/opensearch/templates/securityconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if and .Values.securityConfig.config.securityConfigSecret .Values.securityConfig.config.data }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.securityConfig.config.securityConfigSecret }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: { { .Chart.Name } }
+    release: { { .Release.Name } }
+    heritage: { { .Release.Service } }
```
Comment (bstrdsmkr) [inline charts/opensearch/templates/securityconfig.yaml:10]: This wasn't introduced in this PR, but it's a minor fix that I didn't want to get overlooked in the migration. The extra space between braces causes the chart to fail to parse if you're defining inline configuration in a single secret. Removing the extra space installs appropriately:
```suggestion
    app: {{ .Chart.Name }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
```

Comment (TheAlgo) [reply charts/opensearch/templates/securityconfig.yaml:10]: @bstrdsmkr Thanks for pointing this out. I will make the change.

**Review by smlx — APPROVED** [review]
LGTM. It would be good to get this merged ASAP since it is blocking other PRs.

**Comment by TheAlgo** [comment]
> LGTM. It would be good to get this merged ASAP since it is blocking other PRs.

Agree @smlx . We will be merging it latest by Tuesday(7th September).

**Review by peterzhuamazon — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Pull request merged**